Expensify/App
# Rename and restyle set password form

_**NOTE:** I wasn't able to get git to recognize that the file was renamed, so the diff will show that the `SetPasswordPage` was deleted and that the `SetPasswordForm` is a new file._

### Details
I renamed the `SetPasswordPage` into `SetPasswordForm` (seeing that it only contains the elements relevant to the form now), and changed it to use the same layout/styles as the `LoginForm`. 

I also changed `SignInPageLayout` to use `SafeAreaView` seeing that all the forms who use it need to be wrapped by `SafeAreaView`.

### Fixed Issues
Fixes Expensify/Expensify#157816

### Tests
1. Navigate to the SetPasswordPage
2. Verify that it renders correctly
3. Verify that the set password flow still works.

### QA Steps
1. Navigate to the sign in page. Enter an email address you have access to that doesn't exist. (use [EmailOnDeck](www.emailondeck.com))
2. Click the link in the email that was sent to you.
3. Verify that the set password page renders correctly. Set a password.
4. Sign out, and verify you can sign back in with that password.
5. In addition, to verify that it renders correctly on mobile:
- On native platforms, run `npx uri-scheme open expensify-cash://setpassword/asdf --[ios/android]` once you have the simulator running

### Tested On


### Screenshots

#### Web
<img width="1792" alt="Screen Shot 2021-04-20 at 12 51 06 PM" src="https://user-images.githubusercontent.com/31285285/115339532-85d06600-a1d7-11eb-9154-1d66c9040f8e.png">


#### Mobile Web
<img width="376" alt="Screen Shot 2021-04-20 at 12 51 15 PM" src="https://user-images.githubusercontent.com/31285285/115339543-8e28a100-a1d7-11eb-879a-b3a056824644.png">

#### iOS
![simulator_screenshot_A856EB0A-BD07-4DAD-B63B-0FA9FC9AE969](https://user-images.githubusercontent.com/31285285/115504731-f7c5af80-a2aa-11eb-8517-c12e64c97ead.png)

#### Android
![Screenshot_1618985799](https://user-images.githubusercontent.com/31285285/115505552-36a83500-a2ac-11eb-9690-44bcf4acacdb.png)
